src/requestSchemas.ts

```diff
@@ -0,0 +1,157 @@
+    }
+  } else if (issue.code === z.ZodIssueCode.unrecognized_keys) {
+    throw new BadRequestError(issue.message ?? ctx.defaultError, constants.ISSUE.CODE.VALUE);
+  }
+  throw new BadRequestError(issue.message ?? ctx.defaultError);
+};
+
+z.setErrorMap(customErrorMap);
+
+const hasIdentifyingInfo = (args: Record<string, any>) => args.id || args.url || args.identifier;
+
+/**
+ * Returns a function that checks if any unsupported params are present, then runs the
+ * other passed in functions in sequence. Each catchFunction is expected to check for
+ * invalid input and call ctx.addIssue if invalid input is detected. Any added issue will
+ * trigger an error to be thrown.
+ */
+export function catchInvalidParams(
+  catchFunctions: ((val: Record<string, any>, ctx: z.RefinementCtx) => void)[],
+  unsupportedArgs?: string[]
+) {
+  return (val: Record<string, any>, ctx: z.RefinementCtx) => {
+    const includedUnsupported = Object.keys(val).filter((v: string) => unsupportedArgs?.includes(v));
+    if (includedUnsupported.length > 0) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        params: { type: 'NotImplemented' },
+        message: `Parameters: ${includedUnsupported.join(', ')} are not yet supported on this server`
+      });
+    } else {
+      catchFunctions.forEach(func => func(val, ctx));
+    }
+  };
+}
+
+/**
+ * Checks that either id, url, or identifier parameter is included. Adds an issue to the ctx
+ * that triggers a BadRequest to be thrown if not.
+ */
+export function catchMissingIdentifyingInfo(val: Record<string, any>, ctx: z.RefinementCtx) {
+  if (!hasIdentifyingInfo(val)) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      params: { serverErrorCode: constants.ISSUE.CODE.REQUIRED },
+      message: 'Must provide identifying information via either id, url, or identifier parameters'
+    });
+  }
+}
+
+export const IdentifyingParameters = z
+  .object({
+    id: z.string(),
+    url: z.string().url(),
+    version: z.string(),
+    identifier: z.string()
+  })
+  .partial();
+
+export const PackageArgs = IdentifyingParameters.extend({
+  capability: z.string(),
+  offset: z.string(),
+  count: z.string(),
+  'system-version': z.string(),
+  'check-system-version': z.string(),
+  'force-system-version': z.string(),
+  manifest: z.string(),
+  'include-dependencies': z.string(),
+  'include-components': z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([catchMissingIdentifyingInfo], UNSUPPORTED_PACKAGE_ARGS));
+
+export const DataRequirementsArgs = IdentifyingParameters.extend({
+  expression: z.string(),
+  parameters: z.string(),
+  manifest: z.string(),
+  periodStart: z.string(),
+  periodEnd: z.string(),
+  'include-dependencies': z.string(),
+  'include-components': z.string()
+})
+  .partial()
+  .strict()
+  .superRefine(catchInvalidParams([catchMissingIdentifyingInfo], UNSUPPORTED_DATA_REQ_ARGS));
+
+export const CoreSearchArgs = z
```

Comment:
Do we even want to include parsing of search parameters that the measure repository service doesn't say are SHALLs or SHOULDs?

Reply:
My thinking is that keeping them here will cause the server to throw a 501 instead of a 400 for inclusion of these parameters, which seems slightly more accurate to me. Happy to make the change here if you disagree.

Reply:
I guess it's a question of if we want to make a distinction between "We don't support this right now" and "we will never support this"

I think given the hard limit on resource types that will be supported, it's fine to include the unsupported ones I guess. Maybe there's a world where something changes and we actually want to add support for another search param or something. So it's fine to leave it in for a 501